data-prepper-plugins/grok-prepper/src/test/java/com/amazon/dataprepper/plugins/prepper/grok/GrokPrepperTests.java

```diff
@@ -138,8 +147,131 @@ public void testMatchMerge() throws JsonProcessingException {
         assertThat(equalRecords(grokkedRecords.get(0), resultRecord), equalTo(true));
     }
 
+    @Test
+    public void testBreakOnMatchTrue() throws JsonProcessingException {
+        initialize(getDefaultPluginSetting());
+
+        lenient().when(grok2.match(messageInput)).thenReturn(match2);
+        lenient().when(match2.capture()).thenReturn(capture2);
+
+        capture.put("field_capture_1", "value_capture_1");
+        capture.put("field_capture_2", "value_capture_2");
+        capture.put("field_capture_3", "value_capture_3");
```

Comment:
Does `field_capture_3` play a role in the test? If not, we should remove.

Reply:
It does play a role in this test. I think the naming of `match2` and `grok2` has made this confusing. I will make this more clear with naming.